### PR TITLE
feat: improved editor, add publishing and article view

### PR DESCRIPTION
**Pull request by rin-yato**
## Style
- better darkmode support for editor
- added orange as primary color taken from khmercoder logo

## Api
- added create article api

## Editor 
- added inlinetoolbar
- added drag and drop
- added undo redo

## Other Feature
- able to publish article
- save draft to localstorage, reset on publish
- added simple view article page
- added simple block renderer 

## Fixes
- reset migration
- remove db:generate from README
- fix formatting tailwind classname
- change env.ts to env.mjs
- remove jitti from next config

## Issue
- in the editor, whenever u use the code-editor block and press enter it gets a type error. (only in dev server, its okie in build)
- editor feature is still limited, no image upload, link preview...

### CODE REVIEW

**Comment by soknaly** [comment]
@rin-yato Can you rebase and force push?

**Comment by rin-yato** [comment]
> @rin-yato Can you rebase and force push?

@soknaly done

**Comment by rin-yato** [comment]
@invisal should add us into the vercel preview deployment, i mean grant access

File: src/app/api/article/route.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const article = await db
+    .insert(articles)
+    .values(values.data)
+    .returning()
+    // wrap in result type, this is equivalent to a try catch block
+    .then(ok)
+    .catch(err);
+
+  if (isErr(article))
+    return new Response(article.error.message, { status: 400 });
```
Comment (soknaly) [inline src/app/api/article/route.ts:37]: missing `{ }` 🤣

File: src/app/page.tsx [hunk 2]
```diff
@@ -26,8 +31,20 @@ export default async function HomePage() {
...
+                  key={article.id}
+                  id={article.id}
+                  title={article.title}
+                  description={article.description}
+                  cover={article.cover}
+                  author={{
+                    name: article.author.username,
+                    avatar: article.author.avatar,
+                  }}
+                />
```
Comment (soknaly) [inline src/app/page.tsx:46]: Should we pass the article as props instead of each field as props?

File: src/components/auth/form.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ const AuthForm = ({
       action={action}
       method="post"
       className="mt-4"
-      onSubmit={async (e) => {
+      onSubmit={async e => {
```
Comment (soknaly) [inline src/components/auth/form.tsx:25]: This is the auto-fix? How about we use with bracket all the time?

NOTE: This is just my preference.

File: src/lib/auth/lucia.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ export const auth = lucia({
   env: env.NODE_ENV === "production" ? "PROD" : "DEV",
   middleware: nextjs_future(),
   sessionCookie: { expires: false },
-  getUserAttributes: (data) => {
+  getUserAttributes: data => {
```
Comment (soknaly) [inline src/lib/auth/lucia.ts:20]: Same as the above.

File: src/lib/query/article.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { db } from "../db";
+import { articles } from "../db/schema/articles";
+import { eq } from "drizzle-orm";
+
+export const getAllArticles = async () => {
+  return await db.query.articles.findMany({
+    with: { author: true },
+  });
```
Comment (soknaly) [inline src/lib/query/article.ts:8]: We can remove `await` here?

File: src/app/(private)/write/(components)/publish-article-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    } else {
+      toast.error("Failed to publish article");
+      console.error(result.data);
+    }
+  };
+
+  return (
+    <Dialog modal>
+      <DialogTrigger asChild>{children}</DialogTrigger>
+      <DialogOverlay className="bg-white" />
```
[the remark this answers is not in the document]
Comment (rin-yato) [reply src/app/(private)/write/(components)/publish-article-dialog.tsx]: ✅ done

File: src/app/page.tsx (hunk 2 above)
Comment (rin-yato) [reply src/app/page.tsx:46]: ✅ Done

File: src/components/auth/form.tsx (hunk 3 above)
Comment (rin-yato) [reply src/components/auth/form.tsx:25]: ✅ Done

File: src/lib/auth/lucia.ts (hunk 4 above)
Comment (rin-yato) [reply src/lib/auth/lucia.ts:20]: ✅ Done

File: src/lib/query/article.ts (hunk 5 above)
Comment (rin-yato) [reply src/lib/query/article.ts:8]: ✅ Done

**Comment by rin-yato** [comment]
@soknaly  ✅ Done

**Review by invisal — APPROVED** [review]
LGTM

**Pull request merged**